### PR TITLE
add request tests for basket,products and root

### PR DESCRIPTION
**Pull request by natymaty**

### CODE REVIEW

File: spec/requests/products/show_spec.rb [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+RSpec.describe "GET /products/:id", type: :request do
+  context "without existing product" do
+
+    before do
+      do_request(100)
+    end
+
+    it "returns 404 HTTP code" do
+      p last_response
```
Comment (gotar) [inline spec/requests/products/show_spec.rb]: ???

File: spec/requests/products/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+  context "with existing product" do
+    let (:product) {Shop::FetchProducts.new.call.first}
+
+    before do
+      do_request(product.id)
+    end
+
+    it "returns 200 HTTP code" do
+      p last_response
```
Comment (gotar) [inline spec/requests/products/show_spec.rb]: ??

File: spec/requests/products/show_spec.rb (hunk 1 above)
Comment (natymaty) [inline spec/requests/products/show_spec.rb]: done

File: spec/requests/products/show_spec.rb (hunk 2 above)
Comment (natymaty) [inline spec/requests/products/show_spec.rb]: done

**Pull request merged**